### DISCUSSION TITLE
Explicitly Enable CFI Where Available

### DISCUSSION THREAD

**Discussion by CrossRoast** [Eloston/ungoogled-chromium#3131] [Ideas]
Clang's control flow integrity features seem to be entirely implemented in clang/LLVM, without necessity of operating system support. GN currently enables CFI only in Linux x86 builds. [Vanadium](https://github.com/GrapheneOS/Vanadium/blob/44c6c2c60882569f157731b78d435a2deb504e4b/args.gn) config and [this](https://issues.chromium.org/issues/42200939) Chromium issue imply that CFI does, in fact, work on other platforms.

Why not explicitly enable it everywhere it works for increased memory security in Ungoogled Chromium?

**Comment by PF4Public** [comment]
IIRC, cfi could catch bugs in Chromium codebase specifically. I assume Google does periodic runs and tests with cfi in order to find those bugs, which renders most of them being caught long before us, which in turn would imply very little benefit for us. Am I wrong?

**Reply by CrossRoast** [reply]
Google definitely does fuzzing and all kinds of testing, but you can't be certain there are no bugs. In fact, in such a large codebase, there are very likely some. CFI is meant to be used in production, where the program crashes instead of allowing the attacker to control the process. Its overhead is minimal, < 1%.

**Reply by PF4Public** [reply]
It'd be great to see a report from Vanadium confirming the number of bugs they managed to catch via CFI enablement, which may disprove my statement above. I assume finding a bug in Chromium codebase is a good find that cannot go silent.
